internal/lock/lock.go

```diff
@@ -26,9 +29,25 @@ type Manager struct {
 }
 
 // NewManager returns a new lock manager, ensuring the underlying semaphore is initialized.
-func NewManager(ctx context.Context, etcdURLs []string, group string, slots uint64) (*Manager, error) {
+func NewManager(ctx context.Context, etcdURLs []string, certPubPath string, certKeyPath string, casTrustedPath string, txnTimeoutMs time.Duration, group string, slots uint64) (*Manager, error) {
 	// TODO(lucab): move to clientv3.New(clientv3.Config)
-	client, err := clientv3.NewFromURL(etcdURLs[0])
+
+	tlsInfo := transport.TLSInfo{
+		CertFile:      certPubPath,
+		KeyFile:       certKeyPath,
+		TrustedCAFile: casTrustedPath,
+	}
+
+	tlsConfig, err := tlsInfo.ClientConfig()
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Please drop this, and instead bubble up the error value.

Reply:
Done!!